### ISSUE TITLE
Uncaught TypeError: Cannot read property 'trackPageView' of undefined

### ISSUE THREAD

**Issue by wireframe** [sportly/ember-cli-mixpanel-service#1]
fresh install using the installation guide and I get this error after building:
Uncaught TypeError: Cannot read property 'trackPageView' of undefined

**Comment by kylenathan** [comment]
ember-cli 1.13.1?

**Comment by wireframe** [comment]
correct.  ember env info:

```
Ember-CLI 1.13.1
Ember 1.13.5
Ember Data 1.13.6
jQuery 2.1.4
Ember Simple Auth 0.8.0
```

App config:

```
  ENV['mixpanel'] = {
    enabled: true,
    token: 'my_token',
    LOG_EVENT_TRACKING: true
  };
```

**Comment by kylenathan** [comment]
Thanks, taking a look

**Comment by kylenathan** [comment]
Cant seem to reproduce - message suggests the mixpanel service isn't getting injected into your router for some reason. Two more questions - 

Using version 0.0.5 of this addon?
What other dependencies do you have in your package.json?

**Comment by wireframe** [comment]
package.json

```
  "devDependencies": {
    "broccoli-asset-rev": "^2.0.5",
    "broccoli-funnel": "^0.2.3",
    "ember-cli": "1.13.1",
    "ember-cli-app-version": "^0.3.5",
    "ember-cli-autoprefixer": "0.4.1",
    "ember-cli-babel": "^5.0.0",
    "ember-cli-content-security-policy": "0.4.0",
    "ember-cli-cookie": "^0.1.0",
    "ember-cli-dates": "^1.1.2",
    "ember-cli-dependency-checker": "^1.0.0",
    "ember-cli-dotenv": "^1.0.0",
    "ember-cli-htmlbars": "^0.7.8",
    "ember-cli-ic-ajax": "^0.2.1",
    "ember-cli-inject-live-reload": "^1.3.0",
    "ember-cli-mixpanel-service": "0.0.5",
    "ember-cli-qunit": "^0.3.14",
    "ember-cli-sass": "^4.0.0",
    "ember-cli-simple-auth": "^0.8.0-beta.3",
    "ember-cli-uglify": "^1.0.1",
    "ember-data": "^1.0.0-beta.19",
    "ember-disable-proxy-controllers": "^1.0.0",
    "ember-export-application-global": "^1.0.2",
    "ember-moment": "1.2.1",
    "ember-watson": "^0.5.9",
    "liquid-fire": "^0.21.0",
    "phantomjs": "^1.9.17"
  }
```

**Comment by kylenathan** [comment]
Thanks - will have to some more digging on this.

**Comment by wireframe** [comment]
I think the issue is related to the [instance initializer trying to access the `mixpanel` property on the router when it may not yet be available](https://github.com/sportly/ember-cli-mixpanel-service/blob/master/app/instance-initializers/mixpanel.js#L4).  I did a tweak in my local app and got it to load like this:

``` javascript
export function initialize(instance) {
  var router = instance.container.lookup('router:main');
  router.on('didTransition', function() {
    this.get('container').lookup('service:mixpanel').trackPageView(this.get('url'));
 });
}

export default {
  name: 'mixpanel',
  initialize: initialize
};
```

I'm still fairly new to emberjs, but my understanding is that `this.mixpanel` is not safe and needs to be wrapped in an accessor (ex: `this.get('mixpanel')`) in order to trigger the injections.

does that sound right?

**Comment by walter** [comment]
I'm getting this as well. My approach is slightly different. It makes use of the instance to get the container. Will prepare fork with the fix.

I also wrap it in a conditional to check if mixpanel is enabled, since in development and testing, I disable it. This gives me a bit more leeway with whether I have a token set up or not.

**Comment by walter** [comment]
I'm on Ember 1.13.5 by the way.

**Comment by kylenathan** [comment]
Thanks guys. @walter - fire a pull request and I'll merge.

**Comment by kylenathan** [comment]
Merged and published v0.0.6
